e2e/common/traits/health_test.go

```diff
@@ -363,9 +362,7 @@ func TestHealthTrait(t *testing.T) {
 			name := RandomizedSuffixName("never-ready")
 
 			g.Expect(KamelRunWithID(t, ctx, operatorID, ns, "files/NeverReady.java", "--name", name, "-t", "health.enabled=true",
-				// TODO remove these workaround properties when https://issues.apache.org/jira/browse/CAMEL-20244 is fixed
-				"-p", "camel.route-controller.unhealthyOnRestarting=true",
-				"-p", "camel.route-controller.unhealthyOnExhausted=true",
+				"-p", "camel.health.routesEnabled=false",
```

Comment:
I think this is defeating the goal of the test. The idea is that the health is enabled on Camel side and we use those endpoints to query the route about it's health condition. I am not sure if I am understanding why this option is needed.

Reply:
That is there to avoid any possible (present or future) interaction and side effect between the custom check we define in files/NeverReady.java and the default checks. At the moment the test makes sense and would work both with and without that property, there are no routes declared in files/NeverReady.java anyway...